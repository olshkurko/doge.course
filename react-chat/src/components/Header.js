import React from "react";
import { withStyles } from "material-ui/styles";
import AppBar from "material-ui/AppBar";
import Toolbar from "material-ui/Toolbar";
import Avatar from "./Avatar";
import ChatMenu from "./ChatMenu";
import UserMenu from "./UserMenu";
import Typography from "material-ui/Typography";

const styles = theme => ({
  appBar: {
    position: "fixed",
    width: `calc(100% - 320px)`,
    marginLeft: 320
  },
  appBarTitle: {
    flex: 1,
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit * 2,
    color: theme.palette.secondary.contrastText
  }
});

class Header extends React.Component {
  render() {
    const {
      classes,
      activeUser,
      activeChat,
      logout,
      leaveChat,
      deleteChat,
      editUser
    } = this.props;

    return (
      <AppBar color="primary" className={classes.appBar}>
        <Toolbar color="contrast">
          {activeChat ? (
            <React.Fragment>
              <Avatar colorFrom={activeChat._id}>{activeChat.title}</Avatar>
              <Typography variant="title" className={classes.appBarTitle}>
                {activeChat.title}
                <ChatMenu
                  activeUser={activeUser}
                  onLeaveClick={() => leaveChat(activeChat._id)}
                  onDeleteClick={() => deleteChat(activeChat._id)}
                />
              </Typography>
            </React.Fragment>
          ) : (
            <Typography variant="title" className={classes.appBarTitle}>
              DogeCodes React Chat
            </Typography>
          )}
          <UserMenu
            activeUser={activeUser}
            onLogoutClick={logout}
            onEditProfileClick={editUser}
          />
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styles)(Header);
