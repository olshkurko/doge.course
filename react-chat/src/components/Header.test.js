/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('./Avatar', () => () => 'Avatar');
jest.mock('./ChatMenu', () => () => 'ChatMenu');
jest.mock('./UserMenu', () => () => 'UserMenu');

const mockProps = {
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  activeChat: {
    _id: '12345',
    title: 'My Chat',
  },
  messages: [
    {
      chatId: '12345',
      content: 'content',
      sender: {},
      createdAt: '2018-03-16T10:53:23.200Z',
    },
  ],
  isConnected: true,
  logout: jest.fn(),
  leaveChat: jest.fn(),
  deleteChat: jest.fn(),
  editUser: jest.fn(),
};

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Header {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without active chat', () => {
    const tree = renderer.create(<Header {...mockProps} activeChat={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
