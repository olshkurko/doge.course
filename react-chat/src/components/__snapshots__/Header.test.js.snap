// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header
  className="MuiPaper-root-74 MuiPaper-shadow4-80 MuiAppBar-root-66 MuiAppBar-positionFixed-67 MuiAppBar-colorPrimary-72 mui-fixed Header-appBar-64"
>
  <div
    className="MuiToolbar-root-101 MuiToolbar-gutters-102"
    color="contrast"
  >
    Avatar
    <h2
      className="MuiTypography-root-103 MuiTypography-title-109 Header-appBarTitle-65"
    >
      My Chat
      ChatMenu
    </h2>
    UserMenu
  </div>
</header>
`;

exports[`<Header /> renders without active chat 1`] = `
<header
  className="MuiPaper-root-74 MuiPaper-shadow4-80 MuiAppBar-root-66 MuiAppBar-positionFixed-67 MuiAppBar-colorPrimary-72 mui-fixed Header-appBar-64"
>
  <div
    className="MuiToolbar-root-101 MuiToolbar-gutters-102"
    color="contrast"
  >
    <h2
      className="MuiTypography-root-103 MuiTypography-title-109 Header-appBarTitle-65"
    >
      DogeCodes React Chat
    </h2>
    UserMenu
  </div>
</header>
`;
